apps/server/src/modules/files-storage/service/preview.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Injectable, InternalServerErrorException, NotAcceptableException } from '@nestjs/common';
+import { LegacyLogger } from '@src/core/logger';
+import crypto from 'crypto';
+import { subClass } from 'gm';
+import { PassThrough } from 'stream';
+import { S3ClientAdapter } from '../client/s3-client.adapter';
+import { DownloadFileParams, PreviewParams } from '../controller/dto';
+import { FileRecord } from '../entity';
+import { ErrorType } from '../error';
+import { IGetFile, IGetFileResponse } from '../interface';
+import { PreviewOutputMimeTypes } from '../interface/preview-output-mime-types.enum';
+import { FileDtoBuilder, FileResponseBuilder } from '../mapper';
+import { FilesStorageService } from './files-storage.service';
+
+@Injectable()
+export class PreviewService {
+	constructor(
+		private readonly storageClient: S3ClientAdapter,
+		private readonly fileStorageService: FilesStorageService,
+		private logger: LegacyLogger
+	) {
+		this.logger.setContext(PreviewService.name);
+	}
+
+	private checkIfPreviewPossible(fileRecord: FileRecord): void | NotAcceptableException {
+		if (!fileRecord.isPreviewPossible()) {
+			this.logger.warn(`could not generate preview for : ${fileRecord.id} ${fileRecord.mimeType}`);
+			throw new NotAcceptableException(ErrorType.PREVIEW_NOT_POSSIBLE);
```

Comment:
This definitely not the right exception: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406

I think 422 or 400 would be appropriate.